Update readme

Vinayakj009 opened the pull request:
Changes in the readme have been made to reflect the usage of different functions in the library.

Vinayakj009 committed:
Added download link for zip file

Vinayakj009 committed:
Corrected some typos

Vinayakj009 committed:
Corrected some typos, and added example of how to override global api key and device id.

Vinayakj009 committed:
Added function explanation for begin.

Vinayakj009 committed:
Corrected formatting issues.

Vinayakj009 committed:
Added explanation on how to use hardware serial interface with the begin function.

Vinayakj009 committed:
Added explanation on how to use begin to initialize a software serial port.

Vinayakj009 committed:
Updated readme to include processPushDataCommand explanation.

Vinayakj009 committed:
Added setCommandString explanation to readme.

Vinayakj009 committed:
Added thisng to remember while defining the command fucntion.

Vinayakj009 committed:
Added list of things to remember.

Vinayakj009 committed:
Added explanation for the handle command function.

Vinayakj009 committed:
Made the header file bold

Vinayakj009 committed:
Readme passed through grammerly.

Vinayakj009 committed:
minor corrections.

pranavkundaikar commented on README.md:
```diff
@@ -1,50 +1,155 @@
+To learn more about this function read the example given in File>Examples>BoltIoT-Arduino-Helper>PushData in your Arduino IDE.
+
+## setCommandString
+If you want to use the Bolt Cloud to send commands to the Arduino, or you want to use the Arduino along with the Bolt and the Bolt Cloud to send custom data to your own Cloud, you need to use this function.
```
@Vinayakj009 How does this allow user to send custom data. to their own cloud?

pranavkundaikar commented on README.md:
````diff
@@ -1,50 +1,155 @@
+```
+
+```cpp
+int i=20;
````
@Vinayakj009 Name these variables as sensor_value so that the usage  example will become more relevant.

Vinayakj009 committed:
Update README.md

Made some minor changes.

Vinayakj009 committed:
Update README.md

Added note for number of arguments and number of CSV values for product configuration.

Vinayakj009 committed:
Update README.md

Removed un-necessary info regarding the list.

Vinayakj009 committed:
Add newline.

Added a line at the start of a note.

Vinayakj009 committed:
Gave example for command function.

Vinayakj009 committed:
Noted for setCommandString and handleCommand functions.

Vinayakj009 committed:
Add breakpoints at required places.

Vinayakj009 committed:
Missed breakpoint in earlier commit

Vinayakj009 committed:
Corrected TX pin skip and added extra warning to processPushData function.
